src/structurizr/view/view.py

```diff
@@ -249,10 +273,23 @@ def find_relationship_view(
         self, source_relationship_view: RelationshipView
     ) -> Optional[RelationshipView]:
         """Find a child element view corresponding to the given relationship view."""
-        for relationship_view in self.relationship_views:
+        for relationship_view in self._relationship_views:
             if (
                 relationship_view.relationship.id
                 == source_relationship_view.relationship.id
             ):
                 return relationship_view
         return None
+
+    def check_parent_and_children_not_in_view(self, element: Element) -> None:
+        """Enusre elements can be added with parents or chldren already in the view."""
+        # Unfortunately not every element has a parent property so we have to scan
+        # The children instead.
```

Comment:
Your comment makes me think whether a simple test for the parent attribute makes sense?

```py
if hasattr(element, "parent") and view.element is element.parent:
```

or even

```py
parent = getattr(element, "parent", None)
...
if view.element is parent:
```

Reply:
Yeah, I think that a mix of the two is cleaner than my original:
```python
if view.element is getattr(element, "parent", None):
```